### PR TITLE
Added swaggo operation ID & renamed handlers

### LINKED ISSUE #67: Add Swagger operation IDs to all endpoints
> Based on [RFC-0016](https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup) (https://github.com/iver-wharf/rfcs/pull/16)

Swagger allows for so called "operation IDs", which are string IDs used to identify endpoints.

These IDs are usually used when generating API clients, such as the [swagger-codegen](https://github.com/swagger-api/swagger-codegen) that we use for [wharf-web](https://github.com/iver-wharf/wharf-web).

So these IDs should be descriptive.

## Expected

Names should follow the guidelines specified in https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup#swaggeropenapi-endpoint-ids

Example:

```go
// @id getProjectList
func (m projectModule) getProjectsHandler(c *gin.Context) {}

// @id getProject
func (m projectModule) getProjectHandler(c *gin.Context) {}

// @id createProject
func (m projectModule) postProjectHandler(c *gin.Context) {}
```

To later generate the TypeScript methods:

```ts
ProjectService.getProjectList() {}
ProjectService.getProject(projectId: number) {}
ProjectService.createProject(project: MainProject) {}
```

For consistency, the handler methods in Go should mimic the naming convention, so changing `getProjectsHandler` &rarr; `getProjectListHandler`

## Actual

Swagger-codegen tries to generate its own method names as no IDs exists.

These generated names use the path segments, suffixed with the HTTP method.

Example:

```ts
ProjectService.projectGet() {}

ProjectService.projectsProjectidGet(projectid: number) {}

ProjectService.projectPost(project: MainProject) {}
```

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Added Swaggo `// @id` comment field, which adds Swagger operation IDs to the endpoints.
- Changed name of Gin handler methods to correlate with the new Swagger op IDs

## Motivation

Closes #67

### CODE REVIEW

File: branch.go [hunk 1]
```diff
@@ -19,28 +19,30 @@ type branchModule struct {
 func (m branchModule) Register(g *gin.RouterGroup) {
 	branches := g.Group("/branches")
 	{
-		branches.GET("", m.GetBranchesHandler)
-		branches.PUT("", m.PutBranchesHandler)
+		branches.GET("", m.getBrancheListHandler)
```
Comment (Alexamakans) [inline branch.go:22]: `s/getBrancheListHandler/getBranchListHandler/`

File: CHANGELOG.md [hunk 2]
```diff
@@ -64,6 +64,9 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
   ID has not been shown to have any benefits. Please refer to the
   `GET /project/{projectId}` endpoint instead. (#75)
 
+- Added Swagger operation IDs to all endpoints. This has no effect on the APIs
```
Comment (Alexamakans) [inline CHANGELOG.md:67]:
```suggestion
- Added Swagger operation IDs to all endpoints. This has no effect on the API's
```

**Review by Alexamakans — APPROVED** [review]
LGTM! 👍🏻

**Comment by Alexamakans** [comment]
Wouldn't this potentially count as a MAJOR version bump, since the function names when generating rest clients over at wharf-web will change? Or is it decoupled in a way that makes it not really count?

**Review by iverestefans — APPROVED** [review]
LGTM!

**Comment by jilleJr** [comment]
> Wouldn't this potentially count as a MAJOR version bump, since the function names when generating rest clients over at wharf-web will change? Or is it decoupled in a way that makes it not really count?

The endpoints stay the same. So older versions will still work with the new version.

**Pull request merged**